src/Form/PasswordConstraintDeleteForm.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\password_enhancements\PasswordPolicy;
+use Drupal\user\RoleInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Form for deleting a password constraint.
+ *
+ * @internal
+ */
+final class PasswordConstraintDeleteForm extends ConfirmFormBase {
+
+  /**
+   * The user role containing the password constraint to be deleted.
+   *
+   * @var \Drupal\user\RoleInterface
+   */
+  protected $role;
+
+  /**
+   * The password policy containing the password constraint to be deleted.
+   *
+   * @var \Drupal\password_enhancements\PasswordPolicyInterface
+   */
+  protected $policy;
+
+  /**
+   * The password constraint to be deleted.
+   *
+   * @var \Drupal\password_enhancements\PasswordConstraintInterface
+   */
+  protected $constraint;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Are you sure you want to delete the @constraint constraint from the %role role?', [
+      '%role' => $this->role->label(),
+      '@constraint' => $this->constraint->name(),
+    ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfirmText() {
+    return $this->t('Delete');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return $this->role->toUrl('edit-form');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'password_constraint_delete_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, RoleInterface $user_role = NULL, $password_constraint = NULL) {
+    $this->role = $user_role;
+    // Bail out early if the role doesn't have a policy at all.
+    $this->policy = PasswordPolicy::createFromRole($this->role);
+    if (!$this->policy) {
```

Comment:
Can't we prevent access to this form in a route access handler and make sure we do not print out a "Delete" link/button something that redirects the user to this form that it should have access?
I really not like when we handling "access" in page controllers.